src/backend/lib/constants.js

```diff
@@ -0,0 +1,36 @@
+const DISCOVERY_KEY = "mapeo-upgrade";
+const UPGRADE_INFO_FILENAME = "upgrades.json";
+
+const UpgradeState = {
+  Search: {
+    Idle: 1,
+    Searching: 2,
+    Error: 3,
+  },
+
+  Download: {
+    Idle: 1,
+    Downloading: 2,
+    Downloaded: 3,
+    Error: 4,
+  },
+
+  Check: {
+    NotAvailable: 1,
+    Available: 2,
+    Error: 3,
+  },
+
+  Server: {
+    Idle: 1,
+    Sharing: 2,
+    Draining: 3,
+    Error: 4,
+  },
+};
```

Comment:
This is a style preference, but I prefer strings for state, because it makes debugging easier. If I log the state value then I know what it is, but when it's a number I need to look it up.

Reply:
PS. When I've been writing with Typescript/Flow, then I have dropped using constants, and instead just typed the state variable. Without Typescript I appreciate that using a constant avoids nasty typo errors!